internal/pkg/util/fs/squashfs/squashfs.go

```diff
@@ -0,0 +1,33 @@
+package squashfs
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/config"
+	singularityConfig "github.com/sylabs/singularity/pkg/runtime/engines/singularity/config"
+)
+
+// GetPath figures out where the mksquashfs binary is
+// and return an error is not available or not usable.
+func GetPath() (string, error) {
+	// Parse singularity configuration file
+	c := &singularityConfig.FileConfig{}
+	if err := config.Parser(buildcfg.SYSCONFDIR+"/singularity/singularity.conf", c); err != nil {
```

Comment:
Maybe use `filepath.Join()` here?

Reply:
The code base is full of this problem so I created a new internal function to get the path and created issue #3976 to fix it everywhere.